### PR TITLE
Add client ID to lakeFS client connections

### LINKED ISSUE #68: Add client identification to lakeFS API calls
Identify the Airflow provider with version.

### PR DESCRIPTION
**Pull request by arielshaqed**
Format: "lakefs-airflow-provider (VERSION)".

Fixes #68.

### CODE REVIEW

**Comment by arielshaqed** [comment]
You can see the client string passed [in the provider test](https://github.com/treeverse/airflow-provider-lakeFS/actions/runs/5864320372/job/15899179310) - look for "lakeFS logs", it appears as "`lakefs-airflow-provider (0.46.3)`".

I don't think we should spend time on preventing dev version from misreporting themselves as the latest release.

**Comment by arielshaqed** [comment]
@lynnro314 for Airflow, @nopcoder to check I didn't do something Pythonically stupid.  I will pull with a single approval _unless you ask me not to pull_.

**Review by nopcoder — CHANGES_REQUESTED** [review]
Can we access the version like other python packages:

```python
import lakefs_provider
print(lakefs_provider.__version__)
```

Do we publish the airflow provider like any other package?

File: lakefs_provider/hooks/lakefs_hook.py [hunk 1]
```diff
@@ -31,6 +33,7 @@ class LakeFSHook(BaseHook):
     :type lakefs_conn_id: str
     """
     conn_name_attr = "lakefs_conn_id"
+    client_id = f"lakefs-airflow-provider ({__version__ if __version__ else 'dev'})"
```
Comment (nopcoder) [inline lakefs_provider/hooks/lakefs_hook.py:36]: Suggest: `lakefs-airflow-provider/version`

File: lakefs_provider/version.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+def __get_version(package: str) -> str:
+    try:
+        return metadata.version(package)
+    except metadata.PackageNotFoundError as e:
+        print("No package metadata found")
+        return None
+    except Exception as e:
+        print(f"Get version: {e}")
+        return None
```
Comment (nopcoder) [inline lakefs_provider/version.py:19]: Can we use the standard logging instead of print?

**Review by nopcoder — CHANGES_REQUESTED** [review]

File: lakefs_provider/hooks/lakefs_hook.py (hunk 1 above)
Comment (arielshaqed) [reply lakefs_provider/hooks/lakefs_hook.py]: Done!

Comment (arielshaqed) [reply lakefs_provider/hooks/lakefs_hook.py]: Done, thanks!

File: lakefs_provider/version.py (hunk 2 above)
Comment (arielshaqed) [reply lakefs_provider/version.py]: No longer used, we're doing your method of reading version.

**Comment by arielshaqed** [comment]
Thanks!  I switched to using your method of retrieving the package version.  PTAL...

File: lakefs_provider/version.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
+"""Report package version"""
+
+import sys
```
Comment (nopcoder) [inline lakefs_provider/version.py:3]: not used

File: setup.py [hunk 4]
```diff
@@ -1,14 +1,15 @@
 """Setup.py for the lakeFS Airflow provider package"""
 
 from setuptools import find_packages, setup
+from lakefs_provider.version import __version__
```
Comment (nopcoder) [inline setup.py:4]: from what I understand it is not recommended to import the package from setup.py

File: lakefs_provider/hooks/lakefs_hook.py [hunk 5]
```diff
@@ -1,5 +1,7 @@
 from typing import Any, Dict, IO, Iterator
 
+from lakefs_provider.version import __version__
```
Comment (nopcoder) [inline lakefs_provider/hooks/lakefs_hook.py:3]: can we have it in the package level like other packages?

File: lakefs_provider/hooks/lakefs_hook.py [hunk 6]
```diff
@@ -31,6 +33,7 @@ class LakeFSHook(BaseHook):
     :type lakefs_conn_id: str
     """
     conn_name_attr = "lakefs_conn_id"
+    client_id = f"lakefs-airflow-provider/{__version__ if __version__ else 'dev'}"
```
Comment (nopcoder) [inline lakefs_provider/hooks/lakefs_hook.py:36]: version will be part of the package, no need for a fallback a linter should find it too.

**Review by nopcoder — CHANGES_REQUESTED** [review]

**Review by arielshaqed — COMMENTED** [review]
Thanks!  Changed to yet _another_ version documented by pip.  Ironically this is under the title "single-source".

File: setup.py [hunk 7]
```diff
@@ -1,14 +1,28 @@
 """Setup.py for the lakeFS Airflow provider package"""
 
 from setuptools import find_packages, setup
+from lakefs_provider import __version__
```
Comment (nopcoder) [inline setup.py:4]: remove this one

**Review by nopcoder — CHANGES_REQUESTED** [review]

**Review by nopcoder — APPROVED** [review]
Great, thanks!

**Comment by arielshaqed** [comment]
Thanks!

**Pull request merged**